src/index.ts

```diff
@@ -324,6 +324,35 @@ export const jupyterCommentingPlugin: JupyterFrontEndPlugin<ICommentPanel> = {
 
           const comments = model.comments;
           let index = comments.length;
+          let { start, end } = editorWidget.editor.getSelection();
+          //backwards selection compatibility
+          if (
+            start.line > end.line ||
+            (start.line === end.line && start.column > end.column)
+          ) {
+            [start, end] = [end, start];
+          }
+
+          for (let i = comments.length; i > 0; i--) {
+            const comment = comments.get(i - 1) as ITextSelectionComment;
```

Comment:
This is out-of-bounds for i = 0. Make sure there's a case for when the end (or beginning) of the comments list has been reached.

Reply:
If I'm not mistaken, why would `i` ever be 0 in the loop since the terminating condition is `> 0` not `>= 0`?

Reply:
That's true. In that case , initialize index as 0 instead of comments.length

Reply:
You can also just iterate through starting at 0 and reverse the comparison between the comment position and the selection position